charts/ks-devops/templates/NOTES.txt

```diff
@@ -1,4 +1,11 @@
-1. Get the application URL by running these commands:
+{{if (and (eq .Values.jenkins.Master.ServiceType "NodePort") (not (empty .Values.jenkins.Master.NodePort)))}}
+echo "Visit http://{{.Values.jenkins.Master.HostName}}:{{.Values.jenkins.Master.NodePort}} to use your application"
+{{end}}
+
+1. Get your '{{ .Values.jenkins.Master.AdminUser }}' user password of Jenkins by running:
+  printf $(kubectl get secret --namespace {{ .Release.Namespace }} {{ .Release.Name }}-jenkins -o jsonpath="{.data.jenkins-admin-password}" | base64 --decode);echo
```

Comment:
`printf` should not be here.

Reply:
Why not? It works well if you try to run the command `make install-chart`. You will get the output, and you an get the password with it.

Reply:
Last output and expected output:

```diff
1. Get your 'admin' user password of Jenkins by running:
-  printf $(kubectl get secret --namespace kubesphere-devops-system ks-ctl-jenkins -o jsonpath="{.data.jenkins-admin-password}" | base64 --decode);echo
+ $(kubectl get secret --namespace kubesphere-devops-system ks-ctl-jenkins -o jsonpath="{.data.jenkins-admin-password}" | base64 --decode);echo
```

Reply:
I didn't get your point.

![image](https://user-images.githubusercontent.com/1450685/124694604-36671f80-df14-11eb-995b-4369f18a425e.png)

It looks good to me from above sceenshot.

Reply:
I've updated the codeblock. IMO, the command `printf` is unnecessary from here.

```shell
➜  ~ kubectl get secret --namespace kubesphere-devops-system ks-ctl-jenkins -o jsonpath="{.data.jenkins-admin-password}" | base64 --decode;echo
p101OeIYbNbeEFFVV5NYq9
```

Reply:
> Keyword `printf` should not be here.

`printf` can be here. But I will take your suggestion.

Reply:
See the changes from https://github.com/LinuxSuRen/ks-devops/pull/34/commits/4e3cb7e24125007cdc9036627db3d052f7e3a80c